packages/rulesets/src/spectral/functions/allow-nested-if-parent-exist.ts

```diff
@@ -0,0 +1,53 @@
+// RPC Code: RPC-Get-V1-11
+
+function matchAnyPatterns(patterns: RegExp[], path: string) {
+  return patterns.every((p) => p.test(path))
+}
+
+function verifyNestResourceType(path: string) {
+  const patterns = [/^\/subscriptions\/{subscriptionId}\/resourceGroups\/{resourceGroupName}\/providers\/\w+\.\w+\/\w+\/{\w+}\/\w+.*/gi]
+
+  return matchAnyPatterns(patterns, path)
+}
+
+function verifyResourceType(path: string) {
+  const patterns = [/^\/subscriptions\/{subscriptionId}\/resourceGroups\/{resourceGroupName}\/providers\/\w+\.\w+\/\w+\/{\w+}.*/gi]
+
+  return matchAnyPatterns(patterns, path)
+}
+
+export const allowNestedIfParentExist: any = (fullPath: any, _opts: any, ctx: any) => {
```

Comment:
>allowNestedIfParentExist [](http://example.com/codeflow?start=13&length=24)

The current logic is checking for paths that are under resourceGroups which is fine. Please add a workitem to expand the logic and make this generic for all scopes

Reply:
Essentially we need to detect the api path for a list call for any nested resource type and make sure that the path does not skip any parent segments.